### PR TITLE
Doc updates

### PR DESCRIPTION
**Pull request by rsmmr**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,25 +42,33 @@ parsers, feeding them input, and retrieving their results. At runtime,
...
+that enables adding new protocol and file analyzers to
+[Zeek](https://www.zeek.org) without having to write any C++ code. You
 define the grammar, specify which Zeek events to generate, and Spicy
-takes care of the rest.
+takes care of the rest. There's also a [Zeek
+analyzers](https://github.com/zeek/spicy-analyzers) package that
+provides Zeek with several new, Spicy-based analyzers.
 
+See our [collection of example grammars](https://docs.zeek.org/projects/spicy/en/latest/programming/examples.html#examples)
+to get a sense of how Spicy looks like.
```
Comment (duffy-ocraven) [inline README.md:58]: `to get a sense of how Spicy looks like.` is too colloquial. Suggest:
`to get a sense of how Spicy looks.` or `to get a sense of what Spicy looks like.`

File: README.md [hunk 2]
```diff
@@ -41,25 +42,33 @@ parsers, feeding them input, and retrieving their results. At runtime,
...
 ## Installation
 
-While there are no dedicated releases yet, we provide pre-built Spicy
-binaries for some Linux platforms as well as a Homebrew formula for
-installation on macOS. You can also use one of the included Docker
-files, or just build Spicy from source directly. See the [installation
+We provide pre-built Spicy binaries for several Linux platforms, as
+well as a Homebrew formula (and also binaries) for installation on
+macOS. You can also pull a Docker image from Docker Hub, or leverage
+one of several included Docker files as a starting poing. Of course,
```
Comment (duffy-ocraven) [inline README.md:65]: `point` not `poing`

File: doc/installation.rst [hunk 3]
```diff
@@ -17,109 +30,178 @@ Pre-built binaries
...
+    # tar xf spicy.tar.gz
 
-    brew install zeek  # for Zeek integration
+For JIT support, these binaries require an Xcode installation.
+
+
+.. todo:
+
+    The TAR builds for macOS also don't include the Zeek plugin
+    anymore, correct?
```
Comment (duffy-ocraven) [inline doc/installation.rst:178]: Do we ever put questions in the doc?

File: doc/installation.rst [hunk 4]
```diff
@@ -143,53 +225,91 @@ Install `Docker Desktop for Mac
...
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-You can build your Spicy container from one of the Docker files coming
-with Spicy: Go into Spicy's ``docker`` directory and run ``make`` to
-see the container platforms available::
+You can build base images for your own Spicy setup through the `Docker
+files <https://github.com/zeek/spicy/tree/master/docker>`_ coming with
+the distribution. These images do *not* include Spicy itself, just the
+dependencies that it needs on each platform, both for a source build
+and for the using the corresponding binary packages. (The images do
```
Comment (duffy-ocraven) [inline doc/installation.rst:276]: `for the using the` should be `for using the` (or did you mean something else?)

File: doc/installation.rst [hunk 5]
```diff
@@ -143,53 +225,91 @@ Install `Docker Desktop for Mac
...
+    # cd docker
     # make
 
     Run "make build-<platform>", then "make run-<platform>".
 
     Available platforms:
 
         alpine-3.12
+        centos-7
         centos-8
```
Comment (duffy-ocraven) [inline doc/installation.rst:291]: centos-8 is mentioned here, while above is mentioned:

    * - CentOS 9
      - *(coming soon)

but up there is no mention of CentOS 8, and here no mention of centos-9

File: doc/installation.rst [hunk 6]
```diff
@@ -143,53 +225,91 @@ Install `Docker Desktop for Mac
...
 
-Great, let's fire it up! ::
+.. todo::
 
-    # make run-ubuntu-18
-    root@bc93113300bc:~# spicyc --version
-    0.2.0-dev
+    * The Dockerfiles install different Zeek versions by default. Can
+      we just set them all to the most recent 4.0, and pass different
+      CI versions in through ``ZEEK_VERSION``?
```
Comment (duffy-ocraven) [inline doc/installation.rst:312]: Do we put questions like the one here, into the doc?

File: doc/installation.rst [hunk 7]
```diff
@@ -335,25 +417,26 @@ non-standard location, make sure that ``<prefix>/bin`` is in your
...
 
 .. note::
 
-    By default the cache is located in the folder ``.cache/spicy/<VERSION>`` in
-    the user's home directory. This location can be overriden by setting the
-    environment variable ``SPICY_CACHE`` to a different folder path, both when
-    executing ``spicy-precompile-headers`` and Spicy toolchain commands.
+    By default the cache is located in the folder
+    ``.cache/spicy/<VERSION>`` inside the user's home directory. This
+    location can be overriden by setting the environment variable
```
Comment (duffy-ocraven) [inline doc/installation.rst:440]: `overriden` should be `overridden`

File: doc/zeek.rst [hunk 8]
```diff
@@ -7,63 +7,122 @@ Zeek Integration
...
+
+The easiest way to install the plugin is through Zeek's package
+manager::
+
+    # zkg install zeek/spicy-plugin
+
+This will pull down the plugin's package, compile and test the plugin,
+and then install and activate it. It may take a bit to complete; be
+patient. 🙂 To check that the plugin has become available afterwards,
+run ``zeek -N _Zeek::Spicy``, it should show output like this::
```
Comment (duffy-ocraven) [inline doc/zeek.rst:65]: Should this be one -N or two: -NN

**Comment by rsmmr** [comment]
Thanks, @duffy-ocraven. This is still work in progress, questions will be addressed/removed before merge.

**Comment by rsmmr** [comment]
@bbannier please take a look, there a few TODOs in the installation section, please check them out

File: doc/conf.py [hunk 9]
```diff
@@ -25,7 +25,7 @@
 author = u'Zeek Project'
 
 version = subprocess.check_output("../scripts/autogen-version").decode("utf8")
-release = version
+release = "1.0.0" # most recent release version
```
Comment (bbannier) [inline doc/conf.py:28]: Updating this should end up in a release checklist.

File: doc/conf.py [hunk 10]
```diff
@@ -56,6 +56,14 @@
...
     "issue": ("https://github.com/zeek/spicy/issues/%s", "#"),
     "pr":    ("https://github.com/zeek/spicy/pulls/%s", "#"),
+
+    # Links to binary builds.
+    "package-dev-tgz": ("https://api.cirrus-ci.com/v1/artifact/github/zeek/spicy/%s/packages/spicy.tar.gz", ""),
+    "package-dev-rpm": ("https://api.cirrus-ci.com/v1/artifact/github/zeek/spicy/%s/packages/spicy.rpm", ""),
+    "package-dev-deb": ("https://api.cirrus-ci.com/v1/artifact/github/zeek/spicy/%s/packages/spicy.deb", ""),
+    "package-release-tgz": ("https://api.cirrus-ci.com/v1/artifact/github/zeek/spicy/%%s/packages/spicy-%s.tar.gz" % release, ""),
+    "package-release-rpm": ("https://api.cirrus-ci.com/v1/artifact/github/zeek/spicy/%%s/packages/spicy-%s.rpm" % release, ""),
+    "package-release-deb": ("https://api.cirrus-ci.com/v1/artifact/github/zeek/spicy/%%s/packages/spicy-%s.deb" % release, "")
```
Comment (bbannier) [inline doc/conf.py:66]: This is great.

File: doc/faq.rst [hunk 11]
```diff
@@ -5,6 +5,19 @@
...
 ==========================
 
+Spicy Language
+--------------
+
+.. rubric:: Are Spicy's global variables *really* global?
+
+Indeed, they are. Changes to global variables become visible to all
+Spicy code; their values are not associated with specific connections
+or other dynamic state. This means that often globals won't be the
```
Comment (bbannier) [inline doc/faq.rst:15]: Maybe mention here that they are global to a module.

**Review by bbannier — APPROVED** [review]

File: doc/installation.rst [hunk 12]
```diff
@@ -17,109 +30,192 @@ Pre-built binaries
...
+    # curl -L <link-per-above> -o spicy.tar.gz
+    # tar xf spicy.tar.gz
 
-These binaries require Zeek if you want to use the Spicy Zeek plugin::
+For JIT support, these binaries require an Xcode installation.
 
-    brew install zeek  # for Zeek integration
+.. todo::
+
+    - Do these TAR files need to be extracted into ``/opt/spicy`` as well?
```
[the remark this answers is not in the document]
Comment (rsmmr) [reply doc/installation.rst:190]: Not sure what you mean, TAR files for Linux remain in there (for Alpine at least)?

[the remark this answers is not in the document]
Comment (rsmmr) [reply doc/installation.rst:190]: Oh, I see you mean where we have distribution packages, we don't document the TARs. Yeah, true, but I'd still leave it in for Mac for people not on Homebrew. I'll put in the note on prefix.

File: doc/installation.rst [hunk 13]
```diff
@@ -143,53 +239,93 @@ Install `Docker Desktop for Mac
...
+We provide the following Docker images `on DockerHub <https://hub.docker.com/repository/docker/zeekurity/spicy>`_:
+
+.. list-table::
+    :widths: auto
+    :header-rows: 1
+    :align: center
+
+    * - Spicy Version
+      - Image name
+      - Source
```
[the remark this answers is not in the document]
Comment (rsmmr) [reply doc/installation.rst:256]: `latest` should probably point to the most recent release. Could add a tag `dev` or `latest-dev` to follow nighties.

File: doc/faq.rst (hunk 11 above)
Comment (rsmmr) [reply doc/faq.rst:15]: If they are public, they can be accessed from the outside as well. Is that what you mean?

File: doc/conf.py (hunk 9 above)
Comment (rsmmr) [reply doc/conf.py:28]: https://github.com/zeek/spicy/wiki/Release-Checklist

File: doc/faq.rst (hunk 11 above)
Comment (bbannier) [reply doc/faq.rst:15]: Unless that gets too technical adding that would help.

Is there anything else regarding globals one needs to think of when building a "base module" shared by multiple downstream modules?

**Pull request merged**